### DISCUSSION TITLE
No wifi connection on cold boot

### DISCUSSION THREAD

**Discussion by lin-ycv** [Guilouz/Klipper-Flsun-Speeder-Pad#183] [Q&A] [answered]
I have 3 v400's and it's happening on all of them, on a cold boot, the printer and AP shows an IP, but when i try to ssh to access mainsail, it's unreachable, pining the printer also fails to respond.

If I go into networks and tap on the "connected" wifi again, wifi works.
If I reboot the printer after a cold boot, wifi work.
If I restart klipperscreen after cold boot, wifi works.

I've tried using a bash script and systemd to connect wifi at bootup, but either it's being executed too early, or it's not working.

<details>
<summary>bash script</summary>

```bash
#!/bin/bash

nmcli c up "wifi SSID"
```

</details>

<details>
<summary>systemd</summary>

```
[Unit]
Description=Connect to Wifi

[Service]
TimeoutStartSec=5min
ExecStartPre=/bin/sleep 180
ExecStart=/home/pi/startup.sh

[Install]
WantedBy=default.target
```

</details>

anyone have any ideas?
I think it's a timing issue, but i'm not sure how to solve it.

**Comment by SonOvaSailor** [comment]
I'm having the same problem but I only have one V400 so I just flip this switch off then on and that does it. :)

![20240630_150024](https://github.com/user-attachments/assets/a11584a6-f907-4772-88a0-64da8b51a52f)

**Reply by lin-ycv** [reply]
Yes, but the network page is so laggy, why accept a bug if there's a potential fix?

**Comment by lin-ycv** [comment, marked as answer]
Don't know what the issue is, but here's a workaround
https://youtu.be/SuXGwg9b_1Q

**Reply by bankm1** [reply]
I can confirm this works. I was having the same issue for years on my speeder pads. I use one for each of my FLSUN SR & V400.
